django_client_framework/api/related_model_api.py

```
@@ -106,6 +113,55 @@ def put(self, request, *args, **kwargs):
+        self.__check_write_perm_on_rel_objects(
+            self.field_model.objects.filter(pk__in=self.__body_pk_ls)
+        )
+        if isinstance(self.field, ManyToOneRel):
+            self.__check_write_perm_on_rel_objects(queryset=self.field_val.all())
+            selected = self.field_model.objects.filter(id__in=self.__body_pk_ls)
+            self.field_val.set(selected)
+        else:
+            self.__check_write_perm_on_rel_objects(
+                self.field_model.objects.filter(pk__in=[self.field_val.id])
+            )
+
+            has_read_perm = False
+            if p.has_perms_shortcut(
+                self.user_object, self.model_object, "r", self.field_name
+            ):
+                has_read_perm = True
+
+            new_val_lst = self.field_model.objects.filter(pk__in=self.__body_pk_ls)
+            if len(self.__body_pk_ls) != 1:
+                return Response(
+                    status=400, data={"input_error": "You must input exactly one id"}
+                )
+
+            if len(new_val_lst) == 1:
+                new_val = new_val_lst[0]
+                setattr(self.model_object, self.field_name, new_val)
+                self.model_object.save()
+                if has_read_perm:
+                    # resetting perm because apparently calling self.model_object.save() removes all its
+                    # perms
+                    p.add_perms_shortcut(
+                        self.user_object, self.model_object, "r", self.field_name
+                    )
+                    self.field_val = new_val
+            elif len(new_val_lst) == 0:
+                return Response(
+                    status=400,
+                    data={
+                        "brand_id": 'Invalid pk "{}" - object does not exist.'.format(
+                            self.__body_pk_ls[0]
+                        )
+                    },
+                )
```

Should raise a ValidationError instead of returning a Response